### PR TITLE
main: Exit with error when there is a pending reboot on daemon start

### PR DESCRIPTION
**Pull request by detsch**

### CODE REVIEW

File: src/main.cc [hunk 1]
```diff
@@ -117,7 +117,11 @@ int daemon_main_(LiteClient& client, uint64_t interval, bool return_on_sleep) {
 
   std::shared_ptr<LiteClient> client_ptr{&client, [](LiteClient* /*unused*/) {}};
   AkliteClientExt akclient{client_ptr, false, true};
-  akclient.CompleteInstallation();
+  auto finalize_result = akclient.CompleteInstallation();
```
Comment (mike-sul) [inline src/main.cc:120]: I would check whether there is pending installation before running `CompleteInstallation`:

```
if (client.IsInstallationInProgress) {
   auto finalize_result = akclient.CompleteInstallation();
  ...
}
```

File: src/main.cc [hunk 2]
```diff
@@ -117,7 +117,11 @@ int daemon_main_(LiteClient& client, uint64_t interval, bool return_on_sleep) {
...
+  if (finalize_result.status == InstallResult::Status::NeedsCompletion) {
+    LOG_ERROR << "A system reboot is required to complete a previous installation";
```
Comment (mike-sul) [inline src/main.cc:122]: I would use the terms users have already got used, i.e. "... required to finalize the pending installation."

File: src/main.cc (hunk 1 above)
Comment (detsch) [reply src/main.cc]: Done.

File: src/main.cc (hunk 2 above)
Comment (detsch) [reply src/main.cc]: Done.

**Pull request merged**